anospp_analysis/plasm.py

```diff
@@ -0,0 +1,671 @@
+
+    Parameters:
+    meta_results (pandas.DataFrame): dataframe containing the metadata results
+    run (str): name of the current run
+
+    Returns:
+    None
+    """
+    # Create columns for sorting the dataframe
+    meta_df_all['lims_row'] = meta_df_all.lims_well_id.str.slice(0,1)
+    meta_df_all['lims_col'] = meta_df_all.lims_well_id.str.slice(1).astype(int)
+
+    # Get the lims plate IDs
+    limsplate = meta_df_all.lims_plate_id.unique()
+
+    # # Set up the color map
+    # cmap = {"": "#FFFFFF"}
+    # color_l = list(mcolors.CSS4_COLORS)
+    # for i, species in enumerate(sorted(meta_df_all['plasmodium_species'].astype(str).unique())):
+    #     cmap[species] = mcolors.CSS4_COLORS[color_l[-i]]
+
+    # print(cmap)
+
+    # Make categorical plots for each lims plate
+
+
+    for lims_plate in limsplate:
+        for i in haps_merged_df['target'].unique():
+            if interactive_plotting:
+                from .iplot import plot_plate_view
+                plot_plate_view(meta_df_all[meta_df_all.lims_plate_id == lims_plate].copy(), 
+                                f'{workdir}/plateview_for_{lims_plate}_{i}.html',
+                                i, reference,
+                                f'{lims_plate} Plasmodium positive samples')
+
+        # #Make numerical plots for each lims plate
+        # for i in haps_merged_df['target'].unique():
+        #     plot_lims_plate(meta_df_all[meta_df_all.lims_plate_id == lims_plate].copy(),
+        #                     i, lims_plate,
+        #                     f'{workdir}/plateview_heatmap_{lims_plate}_{i}.png',
+        #                     f'{reference}/species_colours.csv', annot=False)
+
+    # # Make the bar plots
+    # plot_bar(meta_df_all, f'{workdir}/bar_plots.png')
+
+
+
+def plasm(args):
+
+    setup_logging(verbose=args.verbose)
+    os.makedirs(args.outdir, exist_ok=True)
+    os.makedirs(args.workdir, exist_ok=True)
+
+    logging.info('ANOSPP QC data import started')
+    hap_df = prep_hap(args.haplotypes)
+    # plasm_hap_df = hap_df.loc[hap_df.target.isin(PLASM_TARGETS)]  
+    logging.info('## prepare input data and variables')
+    samples_df = prep_samples(args.manifest)
+    stats_df = prep_stats(args.stats)
+    comb_stats_df = combine_stats(stats_df, hap_df, samples_df)
+    # run_id = '_'.join(str(value) for value in comb_stats_df.run_id.unique())
+    haps_merged_df = process_haplotypes(hap_df, comb_stats_df, args.filter_p1, args.filter_p2)
+
+    logging.info('Checking for the presence of PLASM_TARGETS')
+    if len(haps_merged_df['target'].unique()) < 1:
+        logging.warning('Could not find both PLASM_TARGETS in hap_df')
+        sys.exit(1)
+
+    else:
+        logging.info('## run blast')
+        df_list = []
+        hap_output = []
+        for target in haps_merged_df['target'].unique():          
+            haplotype_df, new_cols = haplotype_summary(hap_df, target, args.workdir)
```

Comment:
instead of `Could not find both PLASM_TARGETS in hap_df` check above, do check for `haplotype_df` being empty for a given target

Reply:
suggest prepending per-target variable names with t_ or target_